addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationJob.java

```diff
@@ -133,13 +142,16 @@ protected void applyParameters(
             if (param.getValue() == null) {
                 continue;
             }
-            if (applyCustomParameter(key, param.getValue().toString())) {
+            String resolvedValue;
+            if (env != null) {
+                resolvedValue = env.replaceVars(param.getValue());
+            } else {
+                resolvedValue = param.getValue().toString();
+            }
+            if (applyCustomParameter(key, resolvedValue)) {
                 progress.info(
                         Constant.messages.getString(
-                                "automation.info.setparam",
-                                this.getType(),
-                                key,
-                                param.getValue().toString()));
+                                "automation.info.setparam", this.getType(), key, resolvedValue));
```

Comment:
One important use of env vars is for supplying secrets, eg for authentication.
On that basis I think we should never log the resolved values, only the original ones.
This could make disgnosing problems a bit harder but I dont really see any alternative...